src/prefect/tasks/fivetran/fivetran.py

```diff
@@ -0,0 +1,160 @@
+                if provided this value will override the default polling time of 15 seconds.
+
+        Returns:
+            - dict: connector_id (str) and succeeded_at (timestamp str)
+        """
+
+        def parse_timestamp(api_time: str):
+            """Returns either the pendulum-parsed actual timestamp or
+            a very out-of-date timestamp if not set
+            """
+            return (
+                pendulum.parse(api_time)
+                if api_time is not None
+                else pendulum.from_timestamp(-1)
+            )
+
+        if not connector_id:
+            raise ValueError("Value for parameter `connector_id` must be provided.")
+        if not api_key:
+            raise ValueError("Value for parameter `api_key` must be provided.")
+        if not api_secret:
+            raise ValueError("Value for parameter `api_secret` must be provided.")
+
+        URL_CONNECTOR: str = "https://api.fivetran.com/v1/connectors/{}".format(
+            connector_id
+        )
+
+        self.logger.info(
+            "Attempting start of Fivetran connector {}, sleep time set to {} seconds.".format(
+                connector_id, poll_status_every_n_seconds
+            )
+        )
+
+        # Automatically call `raise_for_status` on every request
+        session = requests.Session()
+        session.hooks = {"response": lambda r, *args, **kwargs: r.raise_for_status()}
+        # Make sure connector configuration has been completed successfully and is not broken.
+        resp = session.get(URL_CONNECTOR, auth=(api_key, api_secret))
+        connector_details = resp.json()["data"]
+        URL_LOGS = "https://fivetran.com/dashboard/connectors/{}/{}/logs".format(
+            connector_details["service"], connector_details["schema"]
+        )
+        URL_SETUP = "https://fivetran.com/dashboard/connectors/{}/{}/setup".format(
+            connector_details["service"], connector_details["schema"]
+        )
+        setup_state = connector_details["status"]["setup_state"]
+        if setup_state != "connected":
+            EXC_SETUP: str = (
+                'Fivetran connector "{}" not correctly configured, status: {}; '
+                + "please complete setup at {}"
+            )
+            raise Exception(EXC_SETUP.format(connector_id, setup_state, URL_SETUP))
+        # We need to know the previous job's completion time to know if the job succeeded or failed
+        succeeded_at = parse_timestamp(connector_details["succeeded_at"])
+        failed_at = parse_timestamp(connector_details["failed_at"])
+        previous_completed_at = succeeded_at if succeeded_at > failed_at else failed_at
+        # URL for connector logs within the UI
+        self.logger.info(
+            "Connector type: {}, connector schema: {}".format(
+                connector_details["service"], connector_details["schema"]
+            )
+        )
+        self.logger.info("Connectors logs at {}".format(URL_LOGS))
```

Comment:
We can reserve this enhancement for v2, but as an FYI note that you can publish "link artifacts" that will get rendered in the UI in a special way; this `URL_LOGS` could be a good candidate for that.  Example code here that publishes a link artifact: https://github.com/PrefectHQ/prefect/blob/master/src/prefect/tasks/prefect/flow_run.py#L191

Reply:
Are there docs for this? I am a little confused how this works just from this code sample.

Reply:
I agree the existing implementation is a little hokey. I'm slightly concerned about them from the POV that Fivetran's UI URLs aren't necessarily a user-facing, supported feature in the way I am using them; they could change without notice.

Reply:
Gotcha, that makes sense; our artifacts API is also [very lightly documented](https://docs.prefect.io/api/latest/artifacts/artifacts.html) at the moment.  For a link artifact, the UI produces a clickable link in a convenient place on the task run / flow run pages for each run.

I'd vote to hold off on this enhancement for now given what you've described 👍